metaflow/plugins/kfp/kfp_cli.py

```diff
@@ -327,6 +334,314 @@ def _convert_value(param: parameters.Parameter):
+        workflow_template: Dict[str, Any] = flow.deploy(
+            kubernetes_namespace, workflow_name
+        )
+        template_name = workflow_template["metadata"]["name"]
+        obj.echo(
+            f"Workflow Template *{template_name}* "
+            f"for flow *{current.flow_name}* pushed to "
+            "Argo Workflows successfully.\n",
+            bold=True,
+        )
+        argo_ui_url = f"{ARGO_RUN_URL_PREFIX}/argo-ui/workflow-templates/{kubernetes_namespace}/{template_name}"
+        obj.echo(f"*Argo:* {argo_ui_url}", fg="cyan")
+
+        # TODO(talebz): using @schedule decorator?
+        #  flow.schedule()
+        #  obj.echo("What will trigger execution of the workflow:", bold=True)
+        #  obj.echo(flow.trigger_explanation(), indent=True)
+
+
+@parameters.add_custom_parameters(deploy_mode=False)
+@kubeflow_pipelines.command(help="Trigger the workflow on Argo Workflows.")
+@click.option(
+    "--name",
+    "name",
+    default=None,
+    help="The workflow template name. The default is the flow name.",
+    show_default=True,
+)
+@click.option(
+    "--k8s-namespace",
+    "--kubernetes-namespace",
+    "kubernetes_namespace",
+    default=KUBERNETES_NAMESPACE,
+    help="Kubernetes Namespace for your run in Argo.",
+    show_default=True,
+)
+@click.option(
+    "--run-id-file",
+    default=None,
+    show_default=True,
+    type=str,
+    help="Write the ID of this run to the file specified.",
+)
+@click.pass_obj
+def trigger(
+    obj,
+    name=None,
+    kubernetes_namespace=KUBERNETES_NAMESPACE,
+    run_id_file=None,
+    **kwargs,
+):
+    def _convert_value(param):
+        # Swap `-` with `_` in parameter name to match click's behavior
+        val = kwargs.get(param.name.replace("-", "_").lower())
+        if param.kwargs.get("type") == JSONType:
+            val = json.dumps(val)
+        return val
+
+    params = {
+        param.name: _convert_value(param)
+        for _, param in obj.flow._get_parameters()
+        if kwargs.get(param.name.replace("-", "_").lower()) is not None
+    }
+
+    # def _convert_value(param: parameters.Parameter):
+    #     v = kwargs.get(param.name)
+    #     return json.dumps(v) if param.kwargs.get("type") == JSONType else v
+    #
+    # flow_parameters = {
+    #     param.name: _convert_value(param)
+    #     for _, param in obj.flow._get_parameters()
+    #     if kwargs.get(param.name) is not None
+    # }
+
+    from metaflow.plugins.kfp.kfp import KubeflowPipelines
+
+    workflow_name: str = resolve_workflow_name(name)
+    workflow: Dict[str, Any] = KubeflowPipelines.trigger(
+        kubernetes_namespace, workflow_name, params
+    )
+    argo_run_id = workflow["metadata"]["name"]
+    metaflow_run_id = to_metaflow_run_id(argo_run_id)
+
+    if run_id_file:
+        with open(run_id_file, "w") as f:
+            f.write(str(argo_run_id))
+
+    obj.echo(
+        f"Workflow *{workflow_name}* triggered on Argo Workflows "
```

Comment:
Do we expect `argo_workflow_name` to differ from `workflow_name` in this function? Also `argo_run_id` seems to be the same as `argo_workflow_name`